src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -142,6 +147,74 @@ public async Task StopContainersAsync(IExecutionContext executionContext, object
             await RemoveContainerNetworkAsync(executionContext, _containerNetwork);
         }
 
+        private async Task<string> GetMSILoginToken(CancellationToken cancellationToken)
+        {
+            // Check environment variable for debugging
+            var envVar = System.Environment.GetEnvironmentVariable("DEBUG_MSI_LOGIN_INFO");
+            // Future: Set this client id. This is the MSI client ID.
+            TokenCredential credential = new ManagedIdentityCredential(clientId: null);
```

Comment:
Potential to optimize allocations by create this conditionally. Use ternary operator to make it concise